test/e2e_node/image.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/dockertools"
+	kubepod "k8s.io/kubernetes/pkg/kubelet/pod"
+)
+
+type ConformanceImage struct {
+	Image   kubecontainer.ImageSpec
+	Runtime kubecontainer.Runtime
+}
+
+func NewConformanceImage(containerRuntime string, image string) (ci ConformanceImage, err error) {
+	ci.Image = kubecontainer.ImageSpec{Image: image}
+	if containerRuntime == "docker" {
+		ci.Runtime = dockerRuntime()
+		return ci, nil
+	}
+	return ci, fmt.Errorf("Unsupported runtime : %s.", containerRuntime)
+}
+
+func dockerRuntime() kubecontainer.Runtime {
+	dockerClient := dockertools.ConnectToDockerOrDie("")
+	pm := kubepod.NewBasicPodManager(nil)
+	dm := dockertools.NewDockerManager(
+		dockerClient,
+		nil, nil, nil, pm, nil,
+		"", 0, 0, "",
+		nil, nil, nil, nil, nil, nil, nil,
+		false, nil, true, false, false,
+	)
+
+	return dm
+}
+
+func (ci *ConformanceImage) Pull() error {
+	err := ci.Runtime.PullImage(ci.Image, nil)
+	if err != nil {
+		return err
+	}
+
+	if present, err := ci.Runtime.IsImagePresent(ci.Image); err != nil {
+		return err
+	} else if present == false {
+		return errors.New("Failed to detect the pulled image.")
+	}
+
+	return nil
+}
+
+func (ci *ConformanceImage) Remove() error {
+	err := ci.Runtime.GarbageCollect(kubecontainer.ContainerGCPolicy{time.Second * 30, 1, 0})
+	if err != nil {
+		return err
+	}
+
+	for start := time.Now(); time.Since(start) < time.Minute*2; time.Sleep(time.Second * 30) {
```

Comment:
I think that is a part of the removing process.
But you are right, the error is not  necessary.